src/riscv.c

```diff
@@ -105,24 +108,112 @@ riscv_word_t rv_get_reg(riscv_t *rv, uint32_t reg)
     return ~0U;
 }
 
-riscv_t *rv_create(const riscv_io_t *io,
-                   riscv_user_t userdata,
-                   int argc,
-                   char **args,
-                   bool output_exit_code)
+/*
+ * Remap standard stream
+ *
+ * @rv: riscv
+ * @fsp: a list of pair of mapping from fd to FILE *
+ * @fsp_size: list size
+ *
+ * Note: fd inside fsp should be 0 or 1 or 2 only
+ *
+ */
+void rv_remap_stdstream(riscv_t *rv, fd_stream_pair_t *fsp, uint32_t fsp_size)
 {
-    assert(io);
+    assert(rv);
+
+    vm_attr_t *attr = PRIV(rv);
+    assert(attr && attr->fd_map);
+
+    for (uint32_t i = 0; i < fsp_size; i++) {
+        int fd = fsp[i].fd;
+        FILE *file = fsp[i].file;
+
+        if (!file)
+            continue;
+        if (fd != STDIN_FILENO && fd != STDOUT_FILENO && fd != STDERR_FILENO)
+            continue;
+
+        /*
+         * check if standard stream refered by fd exists or not
+         */
+        map_iter_t it;
+        map_find(attr->fd_map, &it, &fd);
+        if (it.node) /* found, remove first */
+            map_erase(attr->fd_map, &it);
+        map_insert(attr->fd_map, &fd, &file);
+
+        /*
+         * store new fd to make the vm_attr_t consistent
+         */
+#if !defined(_WIN32) && !defined(_WIN64)
+        int new_fd = fileno(file);
+#else
+        int new_fd = _fileno(file);
+#endif
```

Comment:
How about adding another macro like `FILENO()` here for both, so we can make the codes here clean? Then you can define platform-specific `FILENO()`  at the place you defined things like `STDIN_FILENO`.

Reply:
Got it.

Reply:
Check [commit](https://github.com/sysprog21/rv32emu/pull/340/commits/ea51b2d1300c3d2379c4274af1faa74423fece89#diff-f99b8d1f48ef2257a9bbbe6b487c621f74ef233c00858df5a37b06e89441e836) out.